modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationAuditLogSupport.java

```diff
@@ -0,0 +1,18 @@
+package com.enonic.xp.core.impl.app;
+
+import java.net.URL;
+
+import com.enonic.xp.app.ApplicationKey;
+
+public interface ApplicationAuditLogSupport
+{
+    void startApplication( ApplicationKey applicationKey );
+
+    void stopApplication( ApplicationKey applicationKey );
+
+    void installApplication( ApplicationKey applicationKey, URL url );
+
+    void installApplication( ApplicationKey applicationKey );
```

Comment:
sure you need a separate method? I don't insist, just wonder.

Reply:
We can do it using one method. But I don't like to pass the `null` argument to the method. As for me, in this case the explicit argument is more suitable and a developer will not waste a time to investigate what to do this method.